lib/cdo/aws/s3_packaging.rb

```diff
@@ -149,8 +149,14 @@ def create_package(sub_path)
 
     @logger.info "Attempting to download: #{s3_key}\nto #{package.path}"
     File.open(package, 'wb') do |file|
-      @client.get_object(bucket: BUCKET_NAME, key: s3_key) do |chunk|
-        file.write(chunk)
+      begin
+        @client.get_object(bucket: BUCKET_NAME, key: s3_key) do |chunk|
+          file.write(chunk)
+        end
+      rescue Aws::Errors::MissingCredentialsError, Aws::S3::Errors::ServiceError
+        # Fallback to public URL download if credentials are not provided or invalid.
+        url = Aws::S3::Bucket.new(BUCKET_NAME).object(s3_key).public_url
+        IO.copy_stream open(url), file
```

Comment:
Would there be any downside to just always using the public url instead of having two paths?